### DISCUSSION TITLE
setup MKS ( managed kafka ) consumer for Quarkus

### DISCUSSION THREAD

**Discussion by lalitbyteiq** [quarkusio/quarkus#41088] [Q&A]
previusly i am using self hosted kafka recestly I have moved to mks and I'm facing issue to setup I am authentication for my quarkus application I'm able to make it work on python an springboot can I get some reference on the application.properties how can I setup it

I'm using <dependency> <groupId>io.quarkus</groupId> <artifactId>quarkus-messaging-kafka</artifactId> </dependency>

This is my properties  file

quarkus.http.port=8094

mp.messaging.connector.smallrye-kafka.group.id=listeners-1

kafka.bootstrap.servers = ${KAFKA_SERVER_URL}
kafka.security.protocol = SASL_SSL
kafka.sasl.mechanism = AWS_MSK_IAM
kafka.sasl.jaas.config = software.amazon.msk.auth.iam.IAMLoginModule required;
kafka.sasl.login.callback.handler.class=software.amazon.msk.auth.iam.IAMClientCallbackHandler

mp.messaging.connector.smallrye-kafka.key.deserializer=org.apache.kafka.common.serialization.StringDeserializer
mp.messaging.connector.smallrye-kafka.key.serializer=org.apache.kafka.common.serialization.StringSerializer

mp.messaging.incoming.payments-insights-topic.topic=topic1-${ENVIRONMENT:master}

aws.accessKeyId=<>
aws.secretAccessKey=<>

[a comment by an automated account is not shown]

**Comment by cescoffier** [comment]
What's the exception you are seeing? 

You just need to configure the Kafka client, so it should be vaguely similar to spring boot.

**Reply by lalitbyteiq** [reply]
initially i was getting disconnected but now im getting this 

 failed authentication due to: An error: (java.security.PrivilegedActionException: javax.security.sasl.SaslException: Exception while evaluating challenge [Caused by javax.security.auth.callback.UnsupportedCallbackException: Unrecognized SASL ClientCallback]) occurred when evaluating SASL token received from the Kafka Broker. Kafka Client will go to AUTHENTICATION_FAILED state.

**Reply by cescoffier** [reply]
Can you port the Spring Boot config? It should be very very similar. 

The error message says that `kafka.sasl.login.callback.handler.class=software.amazon.msk.auth.iam.IAMClientCallbackHandler` is invalid. Also, when configuring SASL, white spaces matter....

**Reply by lalitbyteiq** [reply]
quarkus.http.port=8094

mp.messaging.connector.smallrye-kafka.group.id=listeners-1

kafka.bootstrap.servers = ${KAFKA_SERVER_URL}
kafka.security.protocol = SASL_SSL
kafka.sasl.mechanism = AWS_MSK_IAM
kafka.sasl.jaas.config = software.amazon.msk.auth.iam.IAMLoginModule required;
kafka.sasl.login.callback.handler.class=software.amazon.msk.auth.iam.IAMClientCallbackHandler

mp.messaging.connector.smallrye-kafka.key.deserializer=org.apache.kafka.common.serialization.StringDeserializer
mp.messaging.connector.smallrye-kafka.key.serializer=org.apache.kafka.common.serialization.StringSerializer

mp.messaging.incoming.payments-insights-topic.topic=topic1-${ENVIRONMENT:master}

aws.accessKeyId=<>
aws.secretAccessKey=<>

these are the config I'm using

**Reply by cescoffier** [reply]
And what is the spring boot config you are using?

**Reply by ozangunalp** [reply]
@lalitbyteiq Did you make sure you have the 
```
<dependency>
    <groupId>software.amazon.msk</groupId>
    <artifactId>aws-msk-iam-auth</artifactId>
    <version>2.1.0</version>
</dependency>
```
dependency on your application?

**Comment by ozangunalp** [comment]
Ok, I was at last able to reproduce your error. 

You need to configure 
`kafka.sasl.client.callback.handler.class=software.amazon.msk.auth.iam.IAMClientCallbackHandler` (with **client**) 
instead of 
`kafka.sasl.login.callback.handler.class=software.amazon.msk.auth.iam.IAMClientCallbackHandler`.
